Clean-up/clean_lambda.py

```diff
@@ -0,0 +1,62 @@
+"A script to remove unsubscribed products from the database"
+
+from os import _Environ, environ as ENV
+import logging
+import json
+
+from dotenv import load_dotenv
+from psycopg2 import connect
+from psycopg2.extras import RealDictCursor
+from psycopg2.extensions import connection, cursor
+
+
+def get_connection(config: _Environ) -> connection:
+    "Establishes a connection with the database"
+    return connect(
+        user=config["DB_USER"],
+        password=config["DB_PASSWORD"],
+        host=config["DB_HOST"],
+        port=config["DB_PORT"],
+        database=config["DB_NAME"]
+    )
+
+
+def get_cursor(conn: connection) -> cursor:
+    "Returns a cursor for the database"
+    if not isinstance(conn, connection):
+        raise TypeError(
+            'A cursor can only be constructed from a Psycopg2 connection object')
+
+    return conn.cursor(cursor_factory=RealDictCursor)
+
+
+def delete_unsubscribed(conn: connection, table: str) -> list[dict]:
+    "Deletes info about the unsubscribed products from the database"
+    if not isinstance(table, str):
+        raise TypeError('table must be of type string')
+
+    with get_cursor(conn) as cur:
+        cur.execute(f"""
+            DELETE FROM {table}
+            WHERE product_id NOT IN (SELECT product_id FROM subscriptions)
+            RETURNING *;""")
+        data = cur.fetchall()
+    conn.commit()
+    logging.info(data)
```

Comment:
should this be in a try-except block?
```suggestion
try:
    with get_cursor(conn) as cur:
        cur.execute(f"""
            DELETE FROM {table}
            WHERE product_id NOT IN (SELECT product_id FROM subscriptions)
            RETURNING *;""")
        data = cur.fetchall()
    conn.commit()
    logging.info(data)    
except Exception as e:
        logging.error(f"Error deleting from table {table}: {e}")
        conn.rollback()
        raise
```